third_party/terraform/resources/resource_kms_crypto_key.go

```diff
@@ -29,17 +38,30 @@ func resourceKmsCryptoKey() *schema.Resource {
 				Required: true,
 				ForceNew: true,
 			},
+
 			"key_ring": {
 				Type:             schema.TypeString,
 				Required:         true,
 				ForceNew:         true,
 				DiffSuppressFunc: kmsCryptoKeyRingsEquivalent,
 			},
+
 			"rotation_period": {
-				Type:         schema.TypeString,
-				Optional:     true,
-				ValidateFunc: validateKmsCryptoKeyRotationPeriod,
+				Type:             schema.TypeString,
+				Optional:         true,
+				DiffSuppressFunc: kmsCryptoKeyRotationPeriodsEquivalent,
+			},
+
+			"purpose": {
+				Type:             schema.TypeString,
+				ForceNew:         true,
+				Optional:         true,
+				Default:          "encrypt_decrypt",
+				ValidateFunc:     validation.StringInSlice(kmsPurposeNames(), true),
+				StateFunc:        caseLowerStateFunc,
+				DiffSuppressFunc: caseDiffSuppress,
```

Comment:
Given a statefunc, is this necessary? (same below)

Reply:
Yes, for people who already have state from a previous version. In my testing, Terraform applies the StateFunc only on saving, not when comparing. So we need both to prevent silly diffs. It's extra important for these resources since we don't support deletion.